crypto/ocsp/ocsp_http.c

```diff
@@ -16,12 +16,16 @@
 #define OCSP_MAX_RESP_LENGTH    (100 * 1024)
 #define OCSP_MAX_LINE_LEN       4096;
 
-// OCSP states
+// OCSP_REQ_CTX states
 
 // If set no reading should be performed
 #define OHS_NOREAD              0x1000
 // Error condition
 #define OHS_ERROR               (0 | OHS_NOREAD)
+// First call: ready to start I/O
+#define OHS_ASN1_WRITE_INIT     (5 | OHS_NOREAD)
+// Headers set, no final \r\n included
+#define OHS_HTTP_HEADER         (9 | OHS_NOREAD)
```

Comment:
What do each bit of this flag mean? It looks a little weird in binary:
* OHS_NOREAD -> 0b1000000000000
* OHS_ERROR -> 0b1000000000000
* OHS_ASN1_WRITE_INIT -> 0b1000000000101
* OHS_HTTP_HEADER -> 0b1000000001001

Why skip 1 and 2? Why set 5 and 9 which is two bits instead of 4 and 8? Why is noread 4096?

Reply:
These flags were taken from [here](https://github.com/openssl/openssl/blob/OpenSSL_1_1_1-stable/crypto/ocsp/ocsp_ht.c#L39-L60). The list should provide a better picture of everything. I'm skipping the ones that aren't needed for the time being, but all of them are going to eventually be pulled. 
`OCSP_REQ_CTX_nbio` (next PR) uses these flags to determine the read/write state of |OCSP_REQ_CTX|.  `OHS_NOREAD` is a shared state across flags, which `OCSP_REQ_CTX_nbio` uses to check when to start writing to the memory BIO, so it has a more significant bit .